.github/actions/update-best-practice-table/main.py

```diff
@@ -0,0 +1,159 @@
+#! /usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Update the best practices reference doc with the latest practices.
+
+Given clones of the repositories in the charming ecosystem that contain best
+practice notes (in the standard format, in either Markdown or reStructuredText
+files), find all the best practice blocks (and the heading and reference that is
+associated with them) and produce a suitable block of Markdown to include in the
+Ops documentation.
+"""
+
+import argparse
+import pathlib
+import re
+
+
+def extract_best_practice_blocks(file_path: pathlib.Path, content: str):
```

Comment:
A couple of unit tests would be great here.

Or at least two test vectors, RST and MD.